keyboards/melody96/keymaps/zunger/keymap.c

```diff
@@ -0,0 +1,250 @@
+/* Copyright 2019 Yonatan Zunger
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum custom_keycodes {
+  // We provide special layer management keys:
+  //    GREEK triggers the Greek (aka "Front") layer, or the SHIFTGREEK layer when shift is held.
+  //          (Because we use Unicode, we need to implement shift-handling at the firmware level,
+  //          rather than the OS level like we do in the QWERTY layer)
+  //    CADET or GREEK+ALT triggers the Cadet (aka "Top") layer, or the SHIFTCADET layer when
+  //          shift is held.
+  //    LAYER_LOCK locks the "base" layer (i.e., QWERTY, GREEK, or CADET) to the value which is
+  //          pressed at the moment that it is being released. When a layer lock is set, the
+  //          analogous layer modifier key is reversed; e.g., if you lock the GREEK layer, then the
+  //          GREEK button bounces you back to QWERTY.
+  //
+  // We also parse the shift, alt, and caps lock keys to provide management of those which is
+  // compatible with these various layers.
+  KC_GREEK = SAFE_RANGE,
+  KC_CADET,
+  KC_LAYER_LOCK,
+};
+
+enum layers_keymap {
+  _QWERTY = 0,
```

Comment:
```suggestion
  _QWERTY,
```

Reply:
I prefer to err on the side of explicitness with these, especially when using the enums for their int values (as when converting layers to bitfields at the end of process_record_user). Is that OK?

Reply:
I'm not sure what the point would be, considering AFAIK enums always begin at 0.

Reply:
Oh, I know; they compile identically. This is just a matter of being more explicit for clarity to the reader.

Reply:
probably for "human readability"